src/swapd/process_msgs.rs

```diff
@@ -0,0 +1,1040 @@
+use std::{
+    collections::HashMap,
+    time::{Duration, SystemTime},
+};
+
+use crate::{
+    databased::checkpoint_send,
+    event::Event,
+    rpc::{
+        request::{
+            self, BitcoinFundingInfo, CheckpointState, FundingInfo, InitSwap, Msg, Outcome, Params,
+            Reveal, Tx,
+        },
+        Failure, FailureCode, Request, ServiceBus,
+    },
+    swapd::{
+        runtime::Runtime,
+        swap_state::{AliceState, BobState},
+        CheckpointSwapd, SwapCheckpointType,
+    },
+    syncerd::{bitcoin_syncer::p2wpkh_signed_tx_fee, SweepMoneroAddress, XmrAddressAddendum},
+    CtlServer, Endpoints, Error, LogStyle, ServiceId,
+};
+use farcaster_core::{
+    blockchain::Blockchain,
+    role::{SwapRole, TradeRole},
+    swap::btcxmr::message::CoreArbitratingSetup,
+    transaction::TxLabel,
+};
+use microservices::esb::{self, Handler};
+
+use super::{
+    get_swap_id,
+    runtime::{
+        aggregate_xmr_spend_view, remote_params_candidate, PendingRequest, PendingRequests,
+        PendingRequestsT,
+    },
+    swap_state::Awaiting,
+    syncer_client::log_tx_received,
+    State,
+};
+
+impl Runtime {
+    /// Processes incoming RPC or peer requests updating state - and switching to a new state, if
+    /// necessary. Returns bool indicating whether a successful state update happened
+    pub fn process(
+        &mut self,
+        endpoints: &mut Endpoints,
+        source: ServiceId,
+        request: Request,
+    ) -> Result<bool, Error> {
+        let event = Event::with(endpoints, self.identity(), source.clone(), request);
+        let updated_state = match self.process_event(event) {
+            Ok(_) => {
+                // Ignoring possible reporting errors here and after: do not want to
+                // halt the swap just because the client disconnected
+
+                // FIXME: fix report messages
+                // let _ = self.report_progress_message_to(
+                //     endpoints,
+                //     self.enquirer(),
+                //     "", // self.state().state_machine.info_message(swap_id),
+                // );
+
+                true
+            }
+            // We pass ESB errors forward such that they can fail the swap.
+            // In the future they can be caught here and used to re-iterate sending of the same
+            // message later without swap halting.
+            Err(err @ Error::Esb(_)) => {
+                error!(
+                    "{} due to ESB failure: {}",
+                    "Failing swap".err(),
+                    err.err_details()
+                );
+                self.report_failure_to(
```

Comment:
If this (and the one below) is really failing the swap and exiting the runtime, it should report this to farcasterd so farcasterd can do the cleanup. 

If you don't want to fix this here and now, it might even be better to just log and not return an error or exit the runtime in order to not change the failure behavior.